fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountReadPlatformServiceImpl.java

```diff
@@ -1267,6 +1283,136 @@ public Collection<SavingsAccountTransactionData> retrieveAllTransactions(final L
         return this.jdbcTemplate.query(sql, this.transactionsMapper, new Object[] { savingsId, depositAccountType.getValue() }); // NOSONAR
     }
 
+    @Override
+    public Page<SavingsAccountTransactionData> retrieveAllTransactions(final Long savingsId, DepositAccountType depositAccountType,
+            SearchParameters searchParameters) {
+        // validate the search parameters
+        validateSearchParameters(searchParameters);
+
+        Integer page = 0;
+        Integer size = SavingsApiConstants.SAVINGS_ACCOUNT_TRANSACTIONS_DEFAULT_LIMIT;
+        if (searchParameters.isLimited() && searchParameters.isOffset()) {
+            Integer limit = searchParameters.getLimit();
+            page = searchParameters.getOffset() / limit;
+            size = limit;
+        }
+        Pageable pageable = PageRequest.of(page, size);
```

Comment:
What's the point of creating this PageRequest object if you override it in the following if/else?